dipy/denoise/noise_estimate.py

```diff
@@ -274,6 +296,23 @@ def estimate_sigma(arr, disable_background_masking=False):
     k[1, 1, 0] = 1
     k[1, 1, 2] = 1
 
+    # Precomputed factor from Koay 2006, this corrects the bias of magnitude image
+    correction_factor = {1: 0.42920367320510366,
+                         4: 0.4834941393603609,
+                         6: 0.4891759468548269,
+                         8: 0.49195420135894175,
+                        12: 0.4946862482541263,
+                        16: 0.4960339908122364,
+                        20: 0.4968365823718557,
+                        24: 0.49736907650825657,
+                        32: 0.49803177052530145,
+                        64: 0.49901964176235936}
+
+    if N in correction_factor:
+        factor = correction_factor[N]
+    else:
+        factor = correction_factor[1]
```

Comment:
I think that you are forcing a value, by setting the default in the
function signature.

I just meant that if a user called:

    estimate_sigma(arr, N=3)

The function should raise an error, not silently behave as though the user
entered `N=1`.  Or am I missing something?

On Tue, May 19, 2015 at 7:57 AM, Samuel St-Jean <notifications@github.com>
wrote:

> In dipy/denoise/noise_estimate.py
> <https://github.com/nipy/dipy/pull/651#discussion_r30607832>:
>
> > +    # Precomputed factor from Koay 2006, this corrects the bias of magnitude image
> > +    correction_factor = {1: 0.42920367320510366,
> > +                         4: 0.4834941393603609,
> > +                         6: 0.4891759468548269,
> > +                         8: 0.49195420135894175,
> > +                        12: 0.4946862482541263,
> > +                        16: 0.4960339908122364,
> > +                        20: 0.4968365823718557,
> > +                        24: 0.49736907650825657,
> > +                        32: 0.49803177052530145,
> > +                        64: 0.49901964176235936}
> > +
> > +    if N in correction_factor:
> > +        factor = correction_factor[N]
> > +    else:
> > +        factor = correction_factor[1]
>
> I was thinking of raising a warning, but if you don't force any value all
> the old code will break in your face.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/651/files#r30607832>.
>